src/main/java/hudson/tasks/junit/JUnitParser.java

```diff
@@ -104,47 +115,66 @@ public TestResult parseResult(String testResultLocations, Run<?,?> build, FilePa
     }
 
     private static abstract class ParseResultCallable<T> extends MasterToSlaveFileCallable<T> {
-        private final long buildTime;
+        private final long buildStartTimeInMillis;
+
+        private final long buildTimeInMillis;
         private final String testResults;
         private final long nowMaster;
         private final boolean keepLongStdio;
         private final boolean allowEmptyResults;
         private final PipelineTestDetails pipelineTestDetails;
         private final TaskListener listener;
 
+        private final boolean parseOldReports;
+
+        private final boolean debug;
+
         private ParseResultCallable(String testResults, Run<?,?> build,
                                     boolean keepLongStdio, boolean allowEmptyResults,
-                                    PipelineTestDetails pipelineTestDetails, TaskListener listener) {
-            this.buildTime = build.getTimestamp().getTimeInMillis();
+                                    PipelineTestDetails pipelineTestDetails, TaskListener listener,
+                                    boolean parseOldReports, boolean debug) {
+            this.buildStartTimeInMillis = build.getStartTimeInMillis();
+            this.buildTimeInMillis = build.getTimeInMillis();
             this.testResults = testResults;
             this.nowMaster = System.currentTimeMillis();
             this.keepLongStdio = keepLongStdio;
             this.allowEmptyResults = allowEmptyResults;
             this.pipelineTestDetails = pipelineTestDetails;
             this.listener = listener;
+            this.parseOldReports = parseOldReports;
+            this.debug = debug;
         }
 
         @Override
         public T invoke(File ws, VirtualChannel channel) throws IOException {
             final long nowSlave = System.currentTimeMillis();
-
             FileSet fs = Util.createFileSet(ws, testResults);
             DirectoryScanner ds = fs.getDirectoryScanner();
-            TestResult result = null;
-
+            TestResult result;
             String[] files = ds.getIncludedFiles();
             if (files.length > 0) {
-                result = new TestResult(buildTime + (nowSlave - nowMaster), ds, keepLongStdio, pipelineTestDetails);
+                // not sure we can rely seriously on those timestamp so let's take the smaller one...
+                long filesTimestamp = Math.min(buildStartTimeInMillis,buildTimeInMillis);
+                // previous mode buildStartTimeInMillis + (nowSlave - nowMaster);
+                if (debug) {
+                    listener.getLogger().println("ParseResultCallable#invoke buildStartTimeInMillis:" + buildStartTimeInMillis
+                        + ",buildTimeInMillis:" + buildTimeInMillis + ",filesTimestamp:" + filesTimestamp + ",nowSlave:"
+                        + nowSlave + ",nowMaster:" + nowMaster);
+                    }
```

Comment:
Seems like something better sent to `java.util.logging.Logger`?

Reply:
yes definitely not proud of this :)  but I didn't know the `#showAgentLogs` method. This will change my life for sure!
 I will change that.